docs/developer-docs/docs/build-a-keychain/sdk/tutorial.md

```diff
@@ -0,0 +1,409 @@
+---
+sidebar_position: 2
+---
+
+# Building a Keychain service with Warden Protocol
+
+## Understanding the Keychain Service
+
+The Keychain Service is a crucial component in the Warden Protocol ecosystem. It's responsible for managing cryptographic keys and signing operations. Here's a breakdown of its main functions:
+
+1. Key Management:
+   - The service can generate new keys when requested.
+   - It stores and manages these keys securely.
+   - It can provide public keys to other parts of the system that need them.
+
+2. Signing Operations:
+   - When a signing request comes in, the service uses the appropriate private key to sign the provided data.
+   - This allows for secure transactions and operations within the Warden Protocol.
+
+3. Integration with Blockchain:
+   - The service is configured to interact with a specific blockchain (identified by the ChainID).
+   - It uses gRPC to communicate with the blockchain node.
+
+4. Batching and Gas Management:
+   - The service batches operations and manages gas usage to optimize transaction costs and efficiency.
+
+In a production environment, you would need to implement actual key generation and signing logic, integrate with a secure key storage solution, and add more robust error handling and security measures.
```

Comment:
this screams chatgpt all over the place, @ijonele can we summarize and reword this paragraph? I think it's very wordy